homeassistant/components/elmax/config_flow.py

```diff
@@ -181,12 +170,81 @@ async def async_step_panels(
 
     async def async_step_reauth(self, user_input=None):
         """Perform reauth upon an API authentication error."""
+        self._reauth_username = user_input.get(CONF_ELMAX_USERNAME)
+        self._reauth_panelid = user_input.get(CONF_ELMAX_PANEL_ID)
         return await self.async_step_reauth_confirm()
 
     async def async_step_reauth_confirm(self, user_input=None):
         """Handle reauthorization flow."""
-        if user_input is None:
-            return self.async_show_form(step_id="user", data_schema=LOGIN_FORM_SCHEMA)
+        errors = {}
+        if user_input is not None:
+            panel_pin = user_input.get(CONF_ELMAX_PANEL_PIN)
+            password = user_input.get(CONF_ELMAX_PASSWORD)
+            entry = await self.async_set_unique_id(self._reauth_panelid)
+
+            # Handle authentication, make sure the panel we are re-authenticating against is listed among results
+            # and verify its pin is correct.
+            try:
+                # Test login.
+                client = Elmax(username=self._reauth_username, password=password)
+                await client.login()
+
+                # Make sure the panel we are authenticating to is still available.
+                panels = await client.list_control_panels()
+                if (
+                    next(filter(lambda p: p.hash == self._reauth_panelid, panels), None)
```

Comment:
We usually prefer generator functions over filter and lambda

Reply:
Done!